src/arrow1/ffi.rs

```diff
@@ -0,0 +1,181 @@
+use arrow::array::{Array, StructArray};
+use arrow::datatypes::{Field, Schema};
+use arrow::ffi::{self, from_ffi, to_ffi};
+use arrow::record_batch::RecordBatch;
+use wasm_bindgen::prelude::*;
+
+use crate::arrow1::error::Result;
+
+/// Wrapper around an ArrowArray FFI struct in Wasm memory.
+#[wasm_bindgen]
+pub struct FFIArrowArray(Box<ffi::FFI_ArrowArray>);
+
+#[wasm_bindgen]
+impl FFIArrowArray {
+    #[wasm_bindgen]
+    pub fn addr(&self) -> *const ffi::FFI_ArrowArray {
+        self.0.as_ref() as *const _
+    }
+
+    #[wasm_bindgen]
+    pub fn free(self) {
+        drop(self.0)
+    }
+
+    #[wasm_bindgen]
+    pub fn drop(self) {
+        drop(self.0)
+    }
```

Comment:
General note: I need to check that the memory pointed to by this _actually_ gets freed by wasm-bindgen. I only need one method but I'm not sure if I can/need to add a manual `drop` call to release the array